payments/api.py

```diff
@@ -436,3 +438,42 @@ def process_post_sale_webhooks(order_id, source):
             continue
 
         send_post_sale_webhook.delay(system.id, order.id, source)
+
+
+def apply_discount_to_basket(basket_id, discount_id):
+    """
+    Apply a discount to a basket.
+
+    Args:
+        basket_id (int): The ID of the basket to apply the discount to.
+        discount_id (int): The ID of the discount to apply to the basket.
+    """
+    discount = Discount.objects.get(pk=discount_id)
+    basket = Basket.objects.get(pk=basket_id)
```

Comment:
This might be best moved into the Basket model (and the appropriate API call refactored a bit for this).

(In any case you're loading the basket twice - once here and once a couple of lines down.)